Loop/Managers/Alerts/AlertManager.swift

```diff
@@ -88,6 +109,166 @@ public final class AlertManager {
     public func removeAlertSoundVendor(managerIdentifier: String) {
         soundVendors.removeValue(forKey: managerIdentifier)
     }
+
+    // MARK: - Bluetooth alerts
+
+    private func onBluetoothPermissionDenied() {
+        log.default("Bluetooth permission denied")
+        let title = NSLocalizedString("Bluetooth Unavailable Alert", comment: "Bluetooth unavailable alert title")
+        let body = NSLocalizedString("Loop has detected an issue with your Bluetooth settings, and will not work successfully until Bluetooth is enabled. You will not receive glucose readings, or be able to bolus.", comment: "Bluetooth unavailable alert body.")
+        let content = Alert.Content(title: title,
+                                      body: body,
+                                      acknowledgeActionButtonLabel: NSLocalizedString("Dismiss", comment: "Default alert dismissal"))
+        issueAlert(Alert(identifier: bluetoothPoweredOffIdentifier, foregroundContent: content, backgroundContent: content, trigger: .immediate))
+    }
+
+    private func onBluetoothPoweredOn() {
+        log.default("Bluetooth powered on")
+        retractAlert(identifier: bluetoothPoweredOffIdentifier)
+    }
+
+    private func onBluetoothPoweredOff() {
+        log.default("Bluetooth powered off")
+        let title = NSLocalizedString("Bluetooth Off Alert", comment: "Bluetooth off alert title")
+        let bgBody = NSLocalizedString("Loop will not work successfully until Bluetooth is enabled. You will not receive glucose readings, or be able to bolus.", comment: "Bluetooth off background alert body.")
+        let bgcontent = Alert.Content(title: title,
+                                      body: bgBody,
+                                      acknowledgeActionButtonLabel: NSLocalizedString("Dismiss", comment: "Default alert dismissal"))
+        let fgBody = NSLocalizedString("Turn on Bluetooth to receive alerts, alarms or sensor glucose readings.", comment: "Bluetooth off foreground alert body")
+        let fgcontent = Alert.Content(title: title,
+                                      body: fgBody,
+                                      acknowledgeActionButtonLabel: NSLocalizedString("Dismiss", comment: "Default alert dismissal"))
+        issueAlert(Alert(identifier: bluetoothPoweredOffIdentifier,
+                                       foregroundContent: fgcontent,
+                                       backgroundContent: bgcontent,
+                                       trigger: .immediate,
+                                       interruptionLevel: .critical))
+    }
+
+    // MARK: - Loop Not Running alerts
+
+    func loopDidComplete() {
+        clearLoopNotRunningNotifications()
+        scheduleLoopNotRunningNotifications()
+    }
+
+    func scheduleLoopNotRunningNotifications() {
```

Comment:
I know that you are still considering if loop not looping notifications should be stored and retracted. 

In the case this doesn't happen, consider separating out the presentation of the alert with something like
```
func handleAlert(_ alert: Alert) {
    handlers.forEach { $0.issueAlert(alert) }
}
```

such that 

```
public func issueAlert(_ alert: Alert) {
    handleAlert(alert)
    alertStore.recordIssued(alert: alert)
}
```

Then `scheduleLoopNotRunningNotifications` could use the standard presentation of the alert and `inferDeliveredLoopNotRunningNotifications` continue using the standard recording of any presented alert.

Reply:
Added to a comment above, but to improve visibility repeating here.

Based on our discussion shouldn't line 502 become [file]private?

Reply:
Made this private. Hoping we can hold off on larger changes to the way Loop-not-looping alerts work to a different PR, when we've agreed upon the notification changes.